Expensify/App
# Fix condition for creating a Concierge Report

### Details
Previously, we assumed that a user's Concierge report was always the first report we created for them. With the addition of features like Default Rooms and Bulk Compose, this isn't always the case anymore. This assumption was causing accounts in NewDot to be created without a Concierge chat.

This PR fixes this by adding a check for creating the Concierge report.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176228

### Tests
NOTE: To test you need to have https://github.com/Expensify/Web-Expensify/pull/31853 and https://github.com/Expensify/Auth/pull/5903.
1. Create an email using [GuerillaEmail](https://www.guerrillamail.com/). Make sure you use a domain name that doesn't already exist (otherwise we'll create the domain default rooms). Paste it into the `DEV_EMAILS` const in `./Web-Expensify/_config.local.php`.
2. **DON'T** use this email to sign up for an account yet. Using an existing account in NewDot, send a message to that email address. This will allow this email address to already have a report attached to it before its account is created/validated.
3. Now, sign up for an account using this email address in NewDot. Use `vssh php ./Web-Expensify/script/notifyall.php` to send out the email.
4. Click the link provided in the email and set a password. After you're signed in, verify a Concierge chat exists in the LHN.

### QA Steps
1. Create an email using [GuerillaEmail](https://www.guerrillamail.com/). Make sure you use a domain name that doesn't already exist (otherwise we'll create the domain default rooms).
2. **DON'T** use this email to sign up for an account yet. Using an existing account in NewDot, send a message to that email address. This will allow this email address to already have a report attached to it before its account is created/validated.
3. Now, sign up for an account using this email address in NewDot.
4. Click the link provided in the email and set a password. After you're signed in, verify a Concierge chat exists in the LHN.

### Tested On

### Screenshots
N/A
